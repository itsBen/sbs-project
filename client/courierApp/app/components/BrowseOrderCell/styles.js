export default {
  container: {
    flexDirection: 'row',
    paddingVertical: 10,
  },
  image: {
    width: 50,
    height: 50,
    borderRadius: 25,
    resizeMode: 'contain',
  },
  storeTitle: {
    fontWeight: '600',
    fontSize: 17,
    lineHeight: 28,
  },
  text: {
    lineHeight: 28,
  },
  timeText: {
    color: 'red',
  },
  deliveryFee: {
    fontSize: 18,
    textAlign: 'right',
    color: 'green',
  },
};
