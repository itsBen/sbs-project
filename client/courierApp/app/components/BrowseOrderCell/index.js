import React from 'react';
import {
  View,
  Text,
  Image,
  TouchableOpacity,
  TouchableHighlight,
} from 'react-native';
import { Cell } from 'react-native-tableview-simple';
import Icon from 'react-native-vector-icons/MaterialIcons';
import moment from 'moment';
import styles from './styles';
import { getTotalPrice, capitalize } from '@utilities';

const getLogo = storeName => {
  try {
    switch (storeName.toLowerCase()) {
      case 'alepa':
        return require('@assets/logo_alepa.png');
      case 'k-market':
        return require('@assets/logo_kmarket.png');
      case 'lidl':
        return require('@assets/logo_lidl.png');
      default:
        return require('@assets/logo_alepa.png');
    }
  } catch (error) {
    console.log(error);
  }
};

const getTime = timeLimit => {
  return moment(timeLimit).fromNow();
};

export default props => {
  const {
    location,
    timeLimit,
    numberOfItems,
    totalPrice,
    deliveryFee,
    products,
    onPress,
    store,
  } = props;
  const storeName = capitalize(store);

  return (
    <Cell
      cellContentView={
        <TouchableOpacity style={styles.container} onPress={onPress}>
          <View style={{ flex: 1 }}>
            <Image source={getLogo(storeName)} style={styles.image} />
          </View>

          <View style={{ flex: 4, paddingLeft: 12 }}>
            <Text style={styles.storeTitle}>{storeName}</Text>
            <Text style={styles.text}>
              <Icon name="location-on" size={18} />Near {location}
            </Text>
            <Text style={[styles.text, styles.timeText]}>
              <Icon name="access-time" size={18} /> Deliver {getTime(timeLimit)}
            </Text>
            <Text style={styles.text}>
              <Icon name="shopping-cart" size={18} /> {numberOfItems} items
              worth {products && getTotalPrice(products)} €
            </Text>
          </View>
          <View style={{ flex: 2 }}>
            <Text style={styles.deliveryFee}>+ {deliveryFee} €</Text>
          </View>
        </TouchableOpacity>
      }
    />
  );
};
