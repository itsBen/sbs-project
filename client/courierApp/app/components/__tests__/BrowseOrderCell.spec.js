/* eslint-env node, jest */
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import moment from 'moment';
import BrowseOrderCell from '../BrowseOrderCell';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <BrowseOrderCell
        location="Patrick\'s house"
        timeLimit={moment().format()}
        numberOfItems="23"
        totalPrice="34.33"
        deliveryFee="12"
        products={[{ name: 'Milk', price: 2, id: '3443234' }]}
        store="alepa"
        onPress={() => {}}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
