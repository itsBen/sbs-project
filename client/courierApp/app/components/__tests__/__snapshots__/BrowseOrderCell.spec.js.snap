// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "minHeight": 44,
          "paddingLeft": 15,
          "paddingRight": 20,
        },
        Object {
          "backgroundColor": "#FFF",
        },
        Object {},
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flexDirection": "row",
          "opacity": 1,
          "paddingVertical": 10,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Image
          source={undefined}
          style={
            Object {
              "borderRadius": 25,
              "height": 50,
              "resizeMode": "contain",
              "width": 50,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flex": 4,
            "paddingLeft": 12,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 17,
              "fontWeight": "600",
              "lineHeight": 28,
            }
          }
        >
          Alepa
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "lineHeight": 28,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 18,
                },
                undefined,
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
          Near 
          Patrick\\'s house
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "lineHeight": 28,
              },
              Object {
                "color": "red",
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 18,
                },
                undefined,
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
           Deliver 
          a few seconds ago
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "lineHeight": 28,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 18,
                },
                undefined,
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
           
          23
           items worth 
          2.00
           €
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 2,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "green",
              "fontSize": 18,
              "textAlign": "right",
            }
          }
        >
          + 
          12
           €
        </Text>
      </View>
    </View>
  </View>
</View>
`;
